Shopping Cart functionality

safouuwa opened the pull request:
2.7. Build an offline shopping cart that can store multiple tickets.

· The shopping cart should be stored in local storage.

· It should be possible to add and remove tickets from the cart

· The content of the cart must be sent to the reservation endpoint after the user proceeds to checkout.

safouuwa committed:
Shopping Cart functionality

(Non) logged users can now add tickets to a shopping cart, for shopping convenience.

linkverk commented:
Werkt meeste wel goed, maar als er werd "checkout" van cart werkt niet 
![Schermafbeelding 2025-01-08 091534](https://github.com/user-attachments/assets/86d0857e-7e5a-442b-90d8-6a58f2981789)
![Schermafbeelding 2025-01-08 091639](https://github.com/user-attachments/assets/ae66c5d5-6472-4500-bdfd-6a53c2d0853f)

linkverk commented:
Er is bij deze
![Schermafbeelding 2025-01-08 091639](https://github.com/user-attachments/assets/3405bcf6-7eab-492a-b4b8-5489cb5a2e5e)
Er werd als alleen 1 reservation in cart toegevoegd, dan wordt wel met "checkout" gereserveerd. Maar als doen 2 verschillende reservation en dan 1 van 2 verwijderen. Dan er werd die fout met  "Failed to make reservation. Please try again"

safouuwa committed:
Shopping cart Fixes

reservations from the shopping cart are possible again + editing a show that has a reservation placed on it is not possible now.

Also a return button for everywhere needed

safouuwa commented:
Reopen after feedback fixes #1

linkverk commented:
Er is probleem met 2 showtimes. De shopping cart werkt niet met deze
![Schermafbeelding 2025-01-09 190701](https://github.com/user-attachments/assets/c9302f91-8dd8-44c0-841f-2774e160fe6e)
![Schermafbeelding 2025-01-09 190935](https://github.com/user-attachments/assets/95c07716-69af-42ad-8c2f-9dacb03ef41c)

linkverk commented:
Alles werkt wel.
